File: src/main/java/platform/ecommerce/dto/request/product/ProductOptionRequest.java
package platform.ecommerce.dto.request.product;

import jakarta.validation.constraints.*;
import lombok.Builder;
import platform.ecommerce.domain.product.OptionType;

import java.math.BigDecimal;

/**
 * Product option request DTO.
 */
@Builder
public record ProductOptionRequest(

        @NotNull(message = "Option type is required")
        OptionType optionType,

        @NotBlank(message = "Option value is required")
        @Size(max = 100, message = "Option value cannot exceed 100 characters")
        String optionValue,

        @DecimalMin(value = "0", message = "Additional price cannot be negative")
        BigDecimal additionalPrice,

        @Min(value = 0, message = "Stock cannot be negative")
        int stock
) {
}

File: src/main/kotlin/platform/ecommerce/dto/request/ProductOptionRequest.kt
package platform.ecommerce.dto.request

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.PositiveOrZero

data class ProductOptionRequest(
    @field:NotBlank(message = "Option name is required")
    val optionName: String,
    @field:PositiveOrZero(message = "Stock quantity must be zero or greater")
    val stockQuantity: Int
)
